service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/InitiatedCaseEventHandler.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.events.InitiatedCaseEvent;
+import uk.gov.hmcts.reform.fpl.service.LocalAuthorityUserService;
+
+@Component
+public class InitiatedCaseEventHandler {
+
+    private final LocalAuthorityUserService localAuthorityUserService;
+
+    @Autowired
+    public InitiatedCaseEventHandler(LocalAuthorityUserService localAuthorityUserService) {
+        this.localAuthorityUserService = localAuthorityUserService;
+    }
+
+    @Async
+    @EventListener
+    public void handleCaseInitiation(InitiatedCaseEvent event) {
```

Comment:
I think Handlers should communicate what is happening through method signatures. Please rename to `grantCaseAccessToAllLocalAuthorityUsers`